tx-pool/src/service.rs

```diff
@@ -0,0 +1,640 @@
+    pub fn get_block_template(
+        &self,
+        bytes_limit: Option<u64>,
+        proposals_limit: Option<u64>,
+        max_version: Option<Version>,
+    ) -> Result<Box<dyn Future<Item = BlockTemplateResult, Error = oneshot::Canceled>>, FailureError>
+    {
+        let mut sender = self.sender.clone();
+        let (responder, response) = oneshot::channel();
+        let request = Request::call((bytes_limit, proposals_limit, max_version), responder);
+        sender.try_send(Message::BlockTemplate(request))?;
+        Ok(Box::new(response))
+    }
+
+    pub fn notify_new_uncle(&self, uncle: UncleBlockView) -> Result<(), FailureError> {
+        let mut sender = self.sender.clone();
+        let notify = Notify::notify(uncle);
+        sender
+            .try_send(Message::NewUncle(notify))
+            .map_err(Into::into)
+    }
+
+    pub fn update_tx_pool_for_reorg(
+        &self,
+        detached_blocks: VecDeque<BlockView>,
+        attached_blocks: VecDeque<BlockView>,
+        detached_proposal_id: HashSet<ProposalShortId>,
+        snapshot: Arc<Snapshot>,
+    ) -> Result<(), FailureError> {
+        let mut sender = self.sender.clone();
+        let notify = Notify::notify((
+            detached_blocks,
+            attached_blocks,
+            detached_proposal_id,
+            snapshot,
+        ));
+        sender
+            .try_send(Message::ChainReorg(notify))
+            .map_err(Into::into)
+    }
+
+    pub fn submit_txs(
+        &self,
+        txs: Vec<TransactionView>,
+    ) -> Result<Box<dyn Future<Item = SubmitTxsResult, Error = oneshot::Canceled>>, FailureError>
+    {
+        let mut sender = self.sender.clone();
+        let (responder, response) = oneshot::channel();
+        let request = Request::call(txs, responder);
+        sender.try_send(Message::SubmitTxs(request))?;
+        Ok(Box::new(response))
+    }
+
+    pub fn plug_entry(
+        &self,
+        entries: Vec<TxEntry>,
+        target: PlugTarget,
+    ) -> Result<Box<dyn Future<Item = Result<(), ()>, Error = oneshot::Canceled>>, FailureError>
+    {
+        let mut sender = self.sender.clone();
+        let (responder, response) = oneshot::channel();
+        let request = Request::call((entries, target), responder);
+        sender.try_send(Message::PlugEntry(request))?;
+        Ok(Box::new(response))
+    }
+
+    pub fn notify_txs(
+        &self,
+        txs: Vec<TransactionView>,
+        callback: NotifyTxsCallback,
+    ) -> Result<(), FailureError> {
+        let mut sender = self.sender.clone();
+        let notify = Notify::notify((txs, callback));
+        sender
+            .try_send(Message::NotifyTxs(notify))
+            .map_err(Into::into)
+    }
+
+    pub fn get_tx_pool_info(
+        &self,
+    ) -> Result<Box<dyn Future<Item = TxPoolInfo, Error = oneshot::Canceled>>, FailureError> {
+        let mut sender = self.sender.clone();
+        let (responder, response) = oneshot::channel();
```

Comment:
I personally think that it may need to be changed to a synchronized channel, which is crossbeam channel.

Because after `std::future`, future must depend on the runtime, there is no wait for the synchronous blocking execution method, and from the current code, there is no need for an asynchronous channel here, so I think the synchronization channel should be used here.